pkg/sidecar-controller/snapshot_controller.go

```diff
@@ -317,34 +307,66 @@ func (ctrl *csiSnapshotSideCarController) createSnapshotOperation(content *crdv1
 		return content, nil
```

Comment:
I dont think this is needed. createSnapshot will only be called IFF VolumeID != nil && status == nil
thus, introducing the wrapper seems to be redundant.

Reply:
sure

Reply:
The problem of not introducing a createSnapshotWrapper is that we will be calling createSnapshotOperation inside the checkandUpdateContentStatusOperation.

I'm going to get rid of scheduleOperation.  I was going to do that after this PR as it is affecting the requeue logic, but looks like we need to do it now.